File: src/exercises/chapter1/lesson15.rb
# frozen_string_literal: true

# S(n) = 1 + 1/(1) + 2 + 1/(1 + 2) + 3 + 1/(1 + 2 + 3) + … + N + 1/(1 + 2 + 3 + … + N)
# @param {Integer} n - The upper limit of the range
# @return {Float} - The sum of the series where each term includes an integer and its corresponding fractional part

def solution(n)
  result = 0
  denominator = 0

  (1..n).each do |i|
    result += i
    denominator += i
    result += 1.0 / denominator
  end
  result
end

File: src/exercises/chapter1/Lesson15.ts
/**
 * Lesson 15: Calculate S(n) = 1 + 1/1 + 2 + 1/ 1 + 2 + 3 + ….. + 1/ 1 + 2 + 3 + …. + N
 */
export default class Lesson15 {

  /**
   *
   * @param
   * @return
   */

    solution(n: number): number {
      let result = 0

      for (let i = 1; i <= n; i++) {
        let sum = 0
        for (let j = 1; j <= i; j++) {
          sum += j
        }
        result += 1/sum
      }
      return result
    };
}
